etl/steps/data/garden/growth/2022-12-19/gdp_historical.py

```diff
@@ -0,0 +1,133 @@
+    df_wdi = ds["wdi"]
+    df_wdi = pd.DataFrame(df_wdi, dtype=np.float64)
+    df_wdi = df_wdi.reset_index()
+    df_wdi = df_wdi.sort_values(by=["year"]).reset_index(drop=True)
+
+    # Select GDP and GDP pc in international-$ in 2017 prices
+    df_wdi = df_wdi[["country", "year", "ny_gdp_mktp_pp_kd", "ny_gdp_pcap_pp_kd"]]
+    df_wdi = df_wdi.rename(columns={"ny_gdp_mktp_pp_kd": "gdp", "ny_gdp_pcap_pp_kd": "gdp_per_capita"})
+    # Filter "World" entity
+    df_wdi = df_wdi[df_wdi["country"] == "World"]
+    # Drop empty World GDP estimations
+    df_wdi = df_wdi.dropna().reset_index(drop=True)
+
+    # Load Maddison Dataset 2010 - garden step. It contains global GDP estimations between 1 and 1820
+    ds = Dataset(DATA_DIR / f"garden/ggdc/{VERSION_MADDISON2010}/maddison_database")
+    df_maddison2010 = ds["maddison_database"]
+    df_maddison2010 = pd.DataFrame(df_maddison2010)
+
+    # Select variables to evaluate and reference year to compare Maddison world estimations to World Bank WDI (1990)
+    var_list = ["gdp", "gdp_per_capita"]
+    year_ref = 1990
+
+    # Select only "World" entity in Maddison dataframe
+    df_world = df_maddison[df_maddison["country"] == "World"].copy().reset_index(drop=True)
+    df_world = df_world[df_world["year"] <= year_ref].reset_index(drop=True)
+
+    # Drop population, as it's not needed
+    df_world = df_world.drop(columns=["population"])
+
+    # Merge both Maddison world estimation and WDI world estimation
+    df = pd.merge(df_world, df_wdi, on="year", how="left", suffixes=("_maddison", "_wdi"))
+
+    # Apply Maddison Project Database growth (1820-1990) retroactively to 1990 WDI data
+    df = create_estimations_from_growth(df, var_list, year_ref, input_suffix="_maddison", output_suffix="_wdi")
+
+    # Concatenate WDI results with estimation
+    df = pd.concat([df, df_wdi[df_wdi["year"] > year_ref]], ignore_index=True)
+
+    # Specify new reference year for the 1 - 1820 period
+    year_ref = 1820
+
+    # Keep the data until 1820 in the old Maddison Dataset 2010
+    df_maddison2010 = df_maddison2010[df_maddison2010["year"] <= year_ref].reset_index(drop=True)
+    df_maddison2010 = df_maddison2010.drop(columns=["population"])
+
+    # Merge datasets to include Maddison Dataset 2010
+    df = pd.merge(df, df_maddison2010, on="year", how="outer", suffixes=(None, "_maddison2010"), sort=True)
+
+    # Apply Maddison Database 2010 growth (1-1820) retroactively to 1820 estimations
+    df = create_estimations_from_growth(df, var_list, year_ref, input_suffix="_maddison2010", output_suffix="")
+
+    # Round variables to address uncertainty on old estimations (previous to 1990)
+    df["gdp"] = np.where(df["year"] < 1990, df["gdp"].round(-3), df["gdp"])
+    df["gdp_per_capita"] = np.where(df["year"] < 1990, df["gdp_per_capita"].round(-1), df["gdp_per_capita"])
+
+    # create new dataset with the same metadata as meadow
+    ds_garden = Dataset.create_empty(dest_dir)
+
+    tb_garden = Table(df)
```

Comment:
It can be alternatively written as
```
tb_garden = Table(df, short_name="gdp_historical")
ds_garden.add(tb_garden)
ds_garden.update_metadata(N.metadata_path)
ds_garden.save()
```
though it does the same thing.